add opa gatekeeper example for unique names (#132)
awslabs/crossplane-on-eks: examples/gatekeeper/unique-names/README.md

### Block Claim creation that reuses vpcName already managed by another Claim

> This example Gatekeeper policy denies requests for claims that use a vpcName already in used by another Claim.

Reviewer: ```suggestion
This example Gatekeeper policy denies requests for claims with a vpcName already used by another claim
```
